pkg/client/constructor.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"net/http"
+	"sync"
+
+	"github.com/kcp-dev/logicalcluster/v2"
+	"k8s.io/client-go/rest"
+)
+
+type Constructor[R any] struct {
```

Comment:
You want to do godoc for all the things before merging?

Reply:
Sure, can do - but ditto to the rebase cost comment above w.r.t. timelines.